opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -141,7 +143,10 @@ impl<T: Number<T>> Sum<T> {
 
         for (attrs, value) in values.drain() {
             s_data.data_points.push(DataPoint {
-                attributes: attrs,
+                attributes: attrs
+                    .iter()
+                    .map(|(k, v)| KeyValue::new(k.clone(), v.clone()))
+                    .collect(),
```

Comment:
not worried about the extra cost here, as this is in collect() path only, to focus on fixing the hot path first. The collect() path can be refactored afterwards.

Reply:
We can implement `From` trait, and use it throughout?